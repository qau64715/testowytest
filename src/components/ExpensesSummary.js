import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div>
        <p>Expenses count: {props.expensesCount}</p>
        <p>Expenses total: {numeral(props.expensesTotal / 100).format('$0,0.00')}</p>
    </div>
);

const mapStateToProps = (state) => {
    const filteredExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesTotal: selectExpensesTotal(filteredExpenses),
        expensesCount: filteredExpenses.length
    };
};

export default connect(mapStateToProps)(ExpensesSummary);