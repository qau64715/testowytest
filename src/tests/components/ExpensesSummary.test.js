import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount="1" expensesTotal="3222" />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount="0" expensesTotal="0" />);
    expect(wrapper).toMatchSnapshot();
});