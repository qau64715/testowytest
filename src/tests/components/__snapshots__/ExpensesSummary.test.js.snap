// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with expenses 1`] = `
<div>
  <p>
    Expenses count: 
    1
  </p>
  <p>
    Expenses total: 
    $32.22
  </p>
</div>
`;

exports[`should render ExpensesSummary with no expenses 1`] = `
<div>
  <p>
    Expenses count: 
    0
  </p>
  <p>
    Expenses total: 
    $0.00
  </p>
</div>
`;
